tools/ci_build/github/azure-pipelines/templates/use-android-emulator.yml

```
@@ -13,7 +13,22 @@ parameters:
   type: boolean
   default: false
 
+- name: install_missing_packages
+  type: boolean
+  default: false
+
 steps:
+- ${{ if eq(parameters.install_missing_packages, true) }}:
+  - script: |
+      set -e -x
+      ${ANDROID_SDK_ROOT}/cmdline-tools/latest/bin/sdkmanager --install "emulator"
+      ${ANDROID_SDK_ROOT}/cmdline-tools/latest/bin/sdkmanager --install "platform-tools"
+      ${ANDROID_SDK_ROOT}/cmdline-tools/latest/bin/sdkmanager --list
+      ls -R ${ANDROID_SDK_ROOT}/platform-tools
+      echo "##vso[task.setvariable variable=PATH]$PATH:${ANDROID_SDK_ROOT}/platform-tools"
```

this may be problematic with the preceding `set -x`. see this other PR:
https://github.com/microsoft/onnxruntime/pull/15986

```suggestion
      # Do not output ##vso[] commands with `set -x` or they may be parsed again and include a trailing quote.
      set -x
      echo "##vso[task.setvariable variable=PATH]$PATH:${ANDROID_SDK_ROOT}/platform-tools"
```

---

actually, there's also a [prependpath](https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#prependpath-prepend-a-path-to-the--path-environment-variable) task which may be more concise here. it likely has the same issue with `set -x` though.

---

Change will be in this PR #21340